### PR TITLE
[TIMOB-5817] Android: Allow TextField to specify inputType

### PR DESCRIPTION
**Pull request by ashcoding**
Jira: https://jira.appcelerator.org/browse/TIMOB-5817

This PR exposes the inputType (Android only property). This property will force the inputType (and keyboard) in Android to allow for scenarios mentioned in TIMOB-5817.

Test code:

``` javascript
var win = Ti.UI.createWindow({
    modal: true,
    backgroundColor: '#000'
});

var textField = Ti.UI.createTextField({
    width: 200,
    keyboardType: Ti.UI.KEYBOARD_NUMBER_PAD,
    inputType: Ti.UI.INPUT_TYPE_CLASS_NUMBER
});

win.add(textField);

win.open();
```

### CODE REVIEW

**Comment by collinprice** [comment]
Having the inputType field accept an array of INPUT_TYPE_CLASS would have it better mimic the way Android utilizes type classes. This way the array could be ORed together to create a custom keyboard. For example:

~~~javascript
var textField = Ti.UI.createTextField({
    width: 200,
    keyboardType: Ti.UI.KEYBOARD_NUMBER_PAD,
    inputType: [Ti.UI.INPUT_TYPE_CLASS_NUMBER, Ti.UI.INPUT_TYPE_CLASS_TEXT]
});
~~~

File: apidoc/Titanium/UI/TextField.yml [hunk 1]
```diff
@@ -291,7 +291,17 @@ properties:
...
+
+  - name: inputType
+    summary: Input type to accept in thetext field. Also influences the Keyboard type to display.
+    description: |
+        When asking for a specific kind of user input, such as a phone number or email 
+        address, you should always specify the input type that is accepted. This overrides
+        any changes to the keyboard done by the property <Titanium.UI.TextField.keyboardType>.
+    type: Number
+    constants: Titanium.UI.INPUT_TYPE_*
+    platforms: [android]
```
Comment (hieupham007) [inline apidoc/Titanium/UI/TextField.yml]: what's the default value?

**Comment by ashcoding** [comment]
@collinprice That's a good suggestion. Will look into implementing that.

File: apidoc/Titanium/UI/TextField.yml (hunk 1 above)
Comment (ashcoding) [inline apidoc/Titanium/UI/TextField.yml]: Will be adding that.

**Comment by ashcoding** [comment]
Updated example. Now takes in an Array of number.:-
```javascript
var win = Ti.UI.createWindow({
	modal: true,
	backgroundColor: '#000'
});
 
var textField = Ti.UI.createTextField({
	width: 200,
	keyboardType: Ti.UI.KEYBOARD_NUMBER_PAD,
	inputType: [Ti.UI.INPUT_TYPE_CLASS_NUMBER]
});
 
win.add(textField);
 
win.open();
```

**Comment by hieupham007** [comment]
Code reviewed and functionally tested. Request **accepted**

**Pull request merged**